Meadowlark-js/packages/meadowlark-core/src/handler/UriBuilder.ts

```diff
@@ -3,11 +3,37 @@
 // The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
 // See the LICENSE and NOTICES files in the project root for more information.
 
+import { pluralize, uncapitalize } from '@edfi/metaed-plugin-edfi-meadowlark';
 import { PathComponents } from '../model/PathComponents';
+import { FrontendRequest } from './FrontendRequest';
+import { BlockingDocument } from '../message/BlockingDocument';
+import { versionAbbreviationFor } from '../metaed/MetaEdProjectMetadata';
 
 /**
  * Derives the resource URI from the pathComponents and resourceId
  */
 export function resourceUriFrom(pathComponents: PathComponents, resourceId: string): string {
   return `/${pathComponents.version}/${pathComponents.namespace}/${pathComponents.resourceName}/${resourceId}`;
 }
+
+/**
+ * For generating problem details when a document references the document to be deleted.
+ */
+export function blockingDocumentsToUris(frontendRequest: FrontendRequest, blockingDocuments?: BlockingDocument[]): string[] {
+  const result: string[] = [];
+  if (blockingDocuments) {
+    blockingDocuments.forEach((document) => {
+      let uri = resourceUriFrom(
+        {
+          version: versionAbbreviationFor(document.resourceVersion),
+          namespace: document.projectName.toLowerCase(), // Lower casing is correct for Ed-Fi models, not sure about alternatives
```

Comment:
I tried creating a school with ID 124, and then creating a local education agency with ID 124. This should result in a 409. But I'm getting a 500, and there is a log message that I think comes from line 29.

```
2023-02-13 13:25:54.689 error 9e2f02e7-2014-4c37-bbff-a76e729d3011 core.handler.Upsert.upsert 500 TypeError: Cannot read properties of undefined (reading 'toLowerCase')
```